snap/snapcraft.yaml

```diff
@@ -71,8 +71,33 @@ grade: stable
 confinement: classic
 
 parts:
+  # The cryptography library (indirect dependency via craft-store) requires rust, but only ships wheels for amd64 and arm64.
+  # As of cryptography 39, it requires rust v1.48.0 or newer, which is available from focal-updates.
+  # This part contains the full set of build dependencies for cryptography.
+  # NOTE: cryptography requires at least pip version 22 for building.
+  cryptography-deps:
+    plugin: nil
+    build-packages:
+      - cargo
+      - rustc
+      - pkg-config
+      - python3.8-dev
+      - libpython3.8-dev
+      - libssl-dev
+      - libffi-dev
+      - python3-wheel
+      - python3-pip
+      - python3-setuptools-scm
+    override-build: |
+      pip install 'pip>=22.0' 'setuptools>=61.0.0'
```

Comment:
you can probably install this on https://github.com/canonical/charmcraft/pull/1091/files#diff-56759910381a014fecfd7556dd72ddd68c747d922a5b7df2044b9ce7c552f5f5R138